packages/http/src/mocker/negotiator/types.ts

```diff
@@ -14,6 +14,6 @@ export type NegotiationOptions = IHttpOperationConfig;
 
 export type NegotiatePartialOptions = {
   code: string;
-  dynamic: boolean;
+  dynamic: boolean | IHttpOperationDynamicConfig;
```

Comment:
Hmm what is happening here? I generally do not like boolean or something else, but... maybe. 

Also, is this a generic extension system, or a "dynamic mode extension system". I think maybe the former would be more useful, and we pass in things like "mode: dynamic" to that extension system.

Reply:
Well my suggestion here is to build further upon the functionality you've already provided.

3 scenario's:
- Boolean
  - False: Mock with examples.
  - True: Dynamic mocking with file-based extensions from the default directories.
  - IHttpOperationDynamicConfig: Dynamic mocking with programmatically defined extensions +  file-based extensions from the default or reconfigured directories.

The way the code is build now, it's only a "dynamic mode extension system". This suggested extension system plugs in to JsonSchemaFaker's `extend` & `define` methods. 

Does this answer your question?